lib/bolt/result.rb

```diff
@@ -25,6 +25,16 @@ def success?
     end
   end
 
+  class ErrorResult < Result
+    def to_h
+      { 'value' => @message }
+    end
```

Comment:
This seems to be sufficient for the `plan` case:

```
$ bundle exec bolt plan run sample::single_task nodes=qkjbgx7gwjt2dqj.delivery.puppetlabs.net --modules spec/fixtures/modules/
ExecutionResult({'qkjbgx7gwjt2dqj.delivery.puppetlabs.net' => 'Host key verification failed for qkjbgx7gwjt2dqj.delivery.puppetlabs.net: fingerprint ac:8c:26:9b:20:d6:71:2d:cd:a9:1f:40:cb:a9:1f:98 is unknown for "qkjbgx7gwjt2dqj.delivery.puppetlabs.net,10.32.123.243"'})
```

But I don't know if that's sufficient for all cases?

Reply:
but we want it to be an error, this is making it a success in puppet

Reply:
should be something like
```ruby
{
  'error' => {
    'kind' => 'puppetlabs.tasks/connection-error',
    'issue_code' => 'HOST_KEY_SOMETHING',
    'msg' => "Host key verification failed",
    'details' => { ?? }
  }
}
```

Reply:
The `success?` method returns false below, ah but right, the hash contents need to be error like. Does it need to contain `_error` and `_output`, or only `_error`?

Reply:
ah, right you were talking about the issue code and parameterizing it.